server/images.go

```diff
@@ -47,37 +47,35 @@ type Model struct {
 	Options       map[string]interface{}
 }
 
-func (m *Model) Prompt(request api.GenerateRequest) (string, error) {
-	t := m.Template
-	if request.Template != "" {
-		t = request.Template
-	}
-
-	tmpl, err := template.New("").Parse(t)
-	if err != nil {
-		return "", err
-	}
-
-	var vars struct {
-		First  bool
-		System string
-		Prompt string
-	}
+type PromptVars struct {
+	System   string
+	Prompt   string
+	Response string
+}
 
-	vars.First = len(request.Context) == 0
-	vars.System = m.System
-	vars.Prompt = request.Prompt
+// Prompt build a full prompt based on all prompts to this point
+func (m *Model) Prompt(orderedPrompts []PromptVars) (string, error) {
```

Comment:
Why don't we make this:

```suggestion
func (m *Model) Prompt(p Prompt) (string, error) {
```